test/testcafe/framework/page-objects/SelectAuthenticatorPageObject.js

```diff
@@ -2,15 +2,32 @@ import { Selector, userVariables } from 'testcafe';
 import { within } from '@testing-library/testcafe';
 import BasePageObject from './BasePageObject';
 
-const factorListRowSelector = userVariables.gen3 ? '.authenticator-row' : '.authenticator-list .authenticator-row';
+const factorListRowSelector = userVariables.gen3 ? '[data-se="authenticator-button"]' : '.authenticator-list .authenticator-row';
 const factorLabelSelector = `${factorListRowSelector} .authenticator-label`;
-const factorDescriptionSelector = `${factorListRowSelector} .authenticator-description .authenticator-description--text`;
-const factorNicknameSelector = `${factorListRowSelector} .authenticator-description .authenticator-enrollment-nickname`;
-const factorIconSelector = `${factorListRowSelector} .authenticator-icon-container .authenticator-icon`;
-const factorCustomLogoSelector = `${factorListRowSelector} .authenticator-icon-container`;
-const factorSelectButtonDiv = `${factorListRowSelector} .authenticator-button`;
-const factorSelectButtonSelector = `${factorListRowSelector} .authenticator-button .button`;
-const factorUsageTextSelector = `${factorListRowSelector} .authenticator-usage-text`;
+const factorDescriptionSelector = userVariables.gen3
+  ? `${factorListRowSelector} [data-se="authenticator-button-description"]`
+  : `${factorListRowSelector} .authenticator-description .authenticator-description--text`;
+const factorNicknameSelector = userVariables.gen3
+  ? `${factorListRowSelector} [data-se="authenticator-button-nickname"]`
+  : `${factorListRowSelector} .authenticator-enrollment-nickname`;
+const factorIconSelector = userVariables.gen3
+  ? `${factorListRowSelector} [data-se="authenticator-icon"] [data-se="factor-beacon"]`
+  : `${factorListRowSelector} .authenticator-icon-container .authenticator-icon`;
+const factorCustomLogoSelector = userVariables.gen3
+  ? `${factorListRowSelector} [data-se="authenticator-icon"]`
+  : `${factorListRowSelector} .authenticator-icon-container`;
+const factorSelectForVerificationButtonDiv = userVariables.gen3
+  ? `${factorListRowSelector} > div:nth-child(3)`
```

Comment:
I assume this is the best we could get for this selector, right? I'm a bit hesitant on having the selector depend on the exact position of this button div within the component, but if this is what we have to do to get rid of `className` that's fine.

Reply:
Instead we can get `.parentElement` of div with `[data-se="cta-button-icon"]`

Reply:
If this selector is only for the verification code path which, correct me if I'm wrong, is trying to select [this div](https://github.com/okta/okta-signin-widget/blob/93e758241579aa7864b4989cae96ce82c05708b0/src/v3/src/components/AuthenticatorButton/AuthenticatorButton.tsx#L275-L289) then could we switch this to `div:last-child`? That feels a bit safer than specifically `div:nth-child(3)`

Reply:
Yes, it's correct div. `div:last-child` is better, thanks!